.github/workflows/jira-listener.yml

```diff
@@ -0,0 +1,65 @@
+name: Build and push jira-listener container image for Escalation workflow
+
+on:
+  workflow_call:
+  workflow_dispatch:
+  push:
+    branches: [ "main" ]
+    paths:
+      - 'escalation/jira-listener/**'
+      - 'pipeline/**'
+
+env:
+  WORKDIR: escalation/jira-listener
+  MVN_OPTS: ""
+    
+jobs:
+  build-and-push-jira-listener:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+    - name: Build Java projects
+      run: |
+        cd ${WORKDIR}
+        mvn ${MVN_OPTS} -B clean package -DskipTests
+
+    - name: Buildah Action
+      id: build-image
+      uses: redhat-actions/buildah-build@v2
+      with:
+        context: ${{ env.WORKDIR }}
+        image: serverless-workflow-jira-listener
+        tags: latest ${{ github.sha }}
+        extra-args: --ulimit nofile=4096:4096
+        containerfiles: |
+          ${{ env.WORKDIR }}/src/main/docker/Dockerfile.jvm
+
+    - name: Buildah push to OCI Arcive
+      if: ${{ ! inputs.push_pr }}
+      run: |
+        buildah push serverless-workflow-jira-listener:${{ github.sha }} \
+          oci-archive:serverless-workflow-jira-listener-${{ github.sha }}.tar:quay.io/orchestrator/serverless-workflow-jira-listener:${{ github.sha }}
+
+    - name: Save OCI archive
+      if: ${{ ! inputs.push_pr }}
+      uses: actions/upload-artifact@v3
+      with:
+        name: serverless-workflow-jira-listener-${{ github.sha }}.tar
+        path: serverless-workflow-jira-listener-${{ github.sha }}.tar
+
+    - name: Push To quay.io
```

Comment:
general question here, not only related to escalation but to all CI: do we really want to push the image to quay before having the e2e test green? cc @rgolangh

Reply:
ideally no, we need to push post merge. we can have that as a post pr

Reply:
my guess is that it is needed during the e2e run @dmartinol ?

Reply:
in e2e we are downloading the artificact and loading it into the kind registry